### PR TITLE
MH-13083: Unify modal navigation

### PR DESCRIPTION
**Pull request by jaysuz**
(Copied from [Jira](https://opencast.jira.com/browse/MH-13083))
The detail modals in the Admin UI are not consistent considering...

- Title of modal window
- Navigation between individual objects

The goal of this story is to make the detail modals consistent considering the user experience.

### CODE REVIEW

File: modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/groupController.js [hunk 1]
```diff
@@ -128,6 +125,14 @@ angular.module('adminNg.controllers')
           }
         };
 
+        $scope.$on('change', function (event, id) {
+          $scope.$broadcast('clear');
+          $scope.group = GroupResource.get({ id: id }, function () {
+            reloadSelectedRoles();
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/groupController.js]: Switching between groups in the event details window throws a n.role.available.$promise is undefined error and the roles aren't updated.

File: modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js [hunk 2]
```diff
@@ -167,6 +135,62 @@ angular.module('adminNg.controllers')
             }
         };
 
+      /**
+       * This private function updates the scope with the data of a given user.
+       *
+       * @param id the id of the user
+       */
+      function fetchChildResources(id) {
+        console.log($scope.searchField);
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js]: This should be removed.

File: modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js [hunk 3]
```diff
@@ -167,6 +135,62 @@ angular.module('adminNg.controllers')
...
+          Notifications.removeAll('user-form');
+
+          fetchChildResources(id);
+        });
+
+        $scope.$on('clear', function () {
+          $scope.searchFieldExternal = '';
+          $scope.searchFieldEffective = '';
+        });
+
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js:194]: When I create a new user by simply filling out all required fields (so no roles for example), I repeatedly get the error "Error: n.userForm is undefined" for as long as that user exists. I can't reproduce this on develop.opencast.org so this probably has something to do with your changes? No idea with what exactly, though.

**Review by KatrinIhler — CHANGES_REQUESTED** [review]
I tested this and found a couple of things:

File: modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js (hunk 2 above)
Comment (jaysuz) [reply modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js]: Done.

File: modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/groupController.js (hunk 1 above)
Comment (jaysuz) [reply modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/groupController.js]: Addressed with 590a6b83291a6ca8074747e5492e3328dc4c7ce0.

File: modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js (hunk 3 above)
Comment (jaysuz) [reply modules/admin-ui/src/main/webapp/scripts/modules/users/controllers/userController.js:194]: For me, this happens on the develop branch as well, so I don't think this issue is related to the changes I've made. However, I noticed that the "getSubmitButtonState" method is invoked constantly, thus possibly causing this error. When using the ngClass directive as in 48a49bfb6f2d3d07096b69e17175b3214d5f9464, the error does not occur.

**Comment by staubesv** [comment]
H  @gregorydlogan & @lkiesow 

It seems that Travis is in trouble:

> [ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.5:install-node-and-npm (install node and npm) on project opencast-admin-ui: Could not extract the Node archive: Could not extract archive: '/home/travis/.m2/repository/com/github/eirslett/node/8.9.4/node-8.9.4-linux-x64.tar.gz': EOFException -> [Help 1]

I've manually restarted the build but the same error occurred again.

Could you please have a look at why Travis fails?

**Comment by staubesv** [comment]
@KatrinIhler & @lkiesow Seems the ball is currently in your hands... Is there anything left that needs to be addressed?

**Comment by lkiesow** [comment]
@staubesv, no idea until I take another look at this…

**Review by lkiesow — APPROVED** [review]

**Pull request merged**